### DISCUSSION TITLE
Go to definition in a new tab

### DISCUSSION THREAD

**Discussion by stjepangolemac** [zed-industries/zed#36955] [Feature Requests]
### Description

When follow the call stack several through multiple functions using within the same file using "Go to definition" it's hard to keep track of your previous position(s).

### Describe the solution you'd like

A new command that opens the definition in a new tab. It could be an option below the current one, and could also be activated by doing Shift+Cmd+Click on the function name.

**Comment by JosephTLyons** [comment]
Another option, in VS Code, if you `cmd-alt-click` on a variable, it opens a new tab in a new split - https://github.com/zed-industries/feedback/issues/54

**Comment by mightatnight** [comment]
Up

**Comment by mikebronner** [comment]
This seems related, so I'm posting here, as there are many other issues that have since been closed. 

It seems that configuring the behavior to NOT open a new pane, tab, or window, but instead jump the current pane to the definition with option/cmd-click (depending on preferences) does not behave the same as the context menu entry "Go to Definition". When using the mouse/kbd shortcut it will open a split pane, when using the context menu, it updates the current pane. 

I would like to be able to configure this behavior, as many others have differing preferences as is evident by the differing feedback. For me the optimal solution is to jump to the definition in the current pane.

What currently is so annoying that you end up jumping back and forth in panes when you follow multiple definition lookups. VERY annoying.

**Comment by notpeter** [comment]
Apologies for the delayed reply to an old issue.

Currently this is not actually possible.  The default behavior when using goto definition to a different file is to open a new tab, a given buffer cannot be open in multiple tabs within the same pane.

While I understand your use case and have hit the same pain point (navigating 24k lines of [`editor.rs`](https://github.com/zed-industries/zed/blob/main/crates/editor/src/editor.rs) is not fun), supporting opening duplicated tabs is also not particularly intuitive.

There are definitely opportunities for to improve that common flow of tracing the call stack. I wonder if it would be feasible to support a single pane multi-buffer view where where each additional Goto definition spawned as an additional multi-buffer below.

Thanks for filing.
